docs/source/bricks.rst

```diff
@@ -550,6 +593,96 @@ Examples:
   # Returns "Look at me, I'm flying!"
   extract_text_after(text, r"SPEAKER \d{1}:")
 
+``extract_email_address``
+----------------------
+
+Extracts email addresses from a string input and returns a list of all the email
+addresses in the input string.
+
+.. code:: python
+
+  from unstructured.cleaners.extract import extract_email_address
+
+  text = """Me me@email.com and You <You@email.com> 
+      ([ba23::58b5:2236:45g2:88h2]) (10.0.2.01)"""
+
+  # Returns "['me@email.com', 'you@email.com']"
+  extract_email_address(text)
+
+
+``extract_ip_address``
+----------------------
+
+Extracts IPv4 and IPv6 IP addresses in the input string and
+returns a list of all IP address in input string.
+
+.. code:: python
+
+  from unstructured.cleaners.extract import extract_ip_address
+
+   text = """Me me@email.com and You <You@email.com> 
+      ([ba23::58b5:2236:45g2:88h2]) (10.0.2.01)"""
+
+  # Returns "['ba23::58b5:2236:45g2:88h2', '10.0.2.01']"
+ extract_ip_address(text)
+
+
+``extract_ip_address_name``
+----------------------
+
+Extracts the names of each IP address in the ``Received`` field(s) from an ``.eml`` 
+file. ``extract_ip_address_name`` takes in a string and returns a list of all
+IP addresses in the input string.
+
+.. code:: python
+
+  from unstructured.cleaners.extract import extract_ip_address_name
+
+  text = """from ABC.DEF.local ([ba23::58b5:2236:45g2:88h2]) by
+    \n ABC.DEF.local2 ([ba23::58b5:2236:45g2:88h2%25]) with mapi id\
+    n 32.88.5467.123; Fri, 26 Mar 2021 11:04:09 +1200"""
+
+  # Returns "['ABC.DEF.local', 'ABC.DEF.local2']"
+ extract_ip_address_name(text)
```

Comment:
This one runs but the indentation on `extract_ip_address_name` still looks a little off